src/compose/ComposeMenu.js

```diff
@@ -54,69 +76,94 @@ class ComposeMenu extends PureComponent<Props> {
     styles: () => null,
   };
 
-  handleImagePickerResponse = (response: Object) => {
-    if (response.didCancel) {
-      return;
+  handleImageRequest = async (requestType: 'openPicker' | 'openCamera') => {
+    const { dispatch, narrow, onImageSelect, onImageError, maxFiles } = this.props;
+    const defaults = {
+      mediaType: 'photo',
+      compressImageMaxWidth: 2000,
+      compressImageMaxHeight: 2000,
+      forceJpg: true,
+      compressImageQuality: 0.7,
+    };
+    let requestObj = {
+      ...defaults,
+    };
+    let images;
+    if (requestType === 'openPicker') {
+      requestObj = {
+        ...defaults,
+        multiple: true,
+        maxFiles,
+      };
     }
-
-    if (response.error) {
-      showErrorAlert(response.error, 'Error');
+    try {
+      images = await ImagePicker[requestType](requestObj);
+    } catch (e) {
+      onImageError(e);
       return;
     }
-
-    const { dispatch, narrow } = this.props;
-    dispatch(
-      uploadImage(narrow, response.uri, chooseUploadImageFilename(response.uri, response.fileName)),
-    );
+    images = requestType === 'openPicker' ? images : [images];
+    const response = {
+      images: images.map(image => {
+        const inferredFileName = chooseUploadImageFilename(image.path, image.filename);
+        return {
+          uri: image.path,
+          fileName: inferredFileName,
+        };
+      }),
+    };
+    onImageSelect({
+      narrow,
+      response,
+      dispatch,
+    });
   };
-
   handleImageUpload = () => {
-    ImagePicker.launchImageLibrary(
-      {
-        quality: 1.0,
-        noData: true,
-        storageOptions: {
-          skipBackup: true,
-          path: 'images',
-        },
-      },
-      this.handleImagePickerResponse,
-    );
+    this.handleImageRequest('openPicker');
   };
 
   handleCameraCapture = () => {
-    const options = {
-      storageOptions: {
-        cameraRoll: true,
-        waitUntilSaved: true,
-      },
-    };
-
-    ImagePicker.launchCamera(options, this.handleImagePickerResponse);
+    this.handleImageRequest('openCamera');
   };
 
   render() {
     const { styles } = this.context;
-    const { dispatch, expanded, onExpandContract } = this.props;
+    const { dispatch, expanded, onExpandContract, disableUpload, disableCamera } = this.props;
+    let animatedWidth = 40;
```

Comment:
How does this look?
```js
const animatedWidth = 40 + !!disableCamera * 40 + !!disableUpload * 40;
```
Looks a little 'too smart' but also I think not that much, and clearer condensing the logic to an arithmetic calculation (which it kinda is).

Reply:
Looks good. I think I'll use that. Thanks!

Reply:
Actually doesn't work because flow complains about type coercion.